watertap/costing/watertap_costing_package.py

```diff
@@ -857,6 +944,180 @@ def cost_crystallizer_by_volume(blk):
+            doc="Unit adsorbent unit cost per mass",
+        )
+        blk.adsorbent_cost = pyo.Var(
+            initialize=1e5,
+            domain=pyo.NonNegativeReals,
+            units=blk.costing_package.base_currency,
+            doc="Unit adsorbent capital cost",
+        )
+        blk.other_process_cost = pyo.Var(
+            initialize=1e5,
+            domain=pyo.NonNegativeReals,
+            units=blk.costing_package.base_currency,
+            doc="Unit other process capital cost",
+        )
+
+        blk.contactor_cost_constraint = pyo.Constraint(
+            expr=blk.contactor_cost
+            == (
+                blk.costing_package.gac_num_contactors_op
+                + blk.costing_package.gac_num_contactors_redundant
+            )
+            * pyo.units.convert(
+                (
+                    blk.costing_package.gac_contactor_cost_coeff_3
+                    * (
+                        blk.unit_model.bed_volume
+                        / blk.costing_package.gac_num_contactors_op
+                    )
+                    ** 3
+                    + blk.costing_package.gac_contactor_cost_coeff_2
+                    * (
+                        blk.unit_model.bed_volume
+                        / blk.costing_package.gac_num_contactors_op
+                    )
+                    ** 2
+                    + blk.costing_package.gac_contactor_cost_coeff_1
+                    * (
+                        blk.unit_model.bed_volume
+                        / blk.costing_package.gac_num_contactors_op
+                    )
+                    ** 1
+                    + blk.costing_package.gac_contactor_cost_coeff_0
+                ),
+                to_units=blk.costing_package.base_currency,
+            )
+        )
+        blk.bed_mass_gac_ref_constraint = pyo.Constraint(
+            expr=blk.bed_mass_gac_ref
+            == smooth_min(
+                blk.costing_package.bed_mass_gac_max_ref / pyo.units.kg,
+                pyo.units.convert(blk.unit_model.bed_mass_gac, to_units=pyo.units.kg)
+                / pyo.units.kg,
+            )
+            * pyo.units.kg
+        )
+        blk.adsorbent_unit_cost_constraint = pyo.Constraint(
+            expr=blk.adsorbent_unit_cost
+            == pyo.units.convert(
+                blk.costing_package.gac_adsorbent_unit_cost_coeff
+                * pyo.exp(
+                    blk.bed_mass_gac_ref
+                    * blk.costing_package.gac_adsorbent_unit_cost_exp_coeff
+                ),
+                to_units=blk.costing_package.base_currency * pyo.units.kg**-1,
+            )
+        )
+        blk.adsorbent_cost_constraint = pyo.Constraint(
+            expr=blk.adsorbent_cost
+            == blk.adsorbent_unit_cost * blk.unit_model.bed_mass_gac
+        )
+        other_process_cost_units = (
+            pyo.units.m**3
+        ) ** -blk.costing_package.gac_other_cost_exp

```

Comment:
I would just inline this below for clarity.

Reply:
@bknueven Just for clarification, do you mean inline commenting to clarify the "other" (`# other accompanying vessels, pipes, instrumentation, and controls`) or actual code structure changes?

Reply:
No, just replace `other_process_cost_units` below with `(pyo.units.m**3)** - blk.costing_package.gac_other_cost_exp`

Reply:
Got it, more so for conciseness and minimizing declaring intermediate variables. I've gone ahead and made the edit to inline.